tools/clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -7831,6 +7832,12 @@ def err_hlsl_maxrecursiondepth_exceeded : Error<
    "NodeMaxRecursionDepth may not exceed 32">;
 def err_hlsl_too_many_node_inputs : Error<
    "Node shader '%0' may not have more than one input record">;
+def err_hlsl_node_record_type : Error<
+   "%0 is not valid as a node record type">;
+def err_hlsl_node_record_object : Error<
+   "object %0 may not appear in a node record">;
+def err_hlsl_node_record_template : Error<
+   "templated types are not currently supported in a node record">;
```

Comment:
This actually seems extremely limiting and I don't recall this being in the spec. This seems more like a bug than something we should be diagnosing on.

Reply:
It is indeed a bug - or rather several bugs (one with  this implementation, and several in CodeGen) . Until these bugs are fixed I consider the (temporary) diagnostic more friendly than allowing subsequent obscure asserts.

Reply:
Thats fair. I don't think we have an issue tracking that. Can you file one?

Reply:
I have created issue #5729 created to track this issue.